toqito/state_opt/npa_hierarchy.py

```diff
@@ -6,119 +6,225 @@
+
+        for word_a_tuple in product(alice_symbols, repeat=alice_len_conf):
+            reduced_a = _reduce(word_a_tuple)
+            if reduced_a == () and alice_len_conf > 0:
+                continue
+
+            for word_b_tuple in product(bob_symbols, repeat=bob_len_conf):
+                reduced_b = _reduce(word_b_tuple)
+                if reduced_b == () and bob_len_conf > 0:
+                    continue
+
+                # Combine and add as in the main loop
+                # Both parts are empty (e.g. alice_len_conf=0, bob_len_conf=0)
+                if not reduced_a and not reduced_b:
+                    # Should not happen if _parse filters (0,0) from conf
+                    final_word = (IDENTITY_SYMBOL,)
+
+                else:
+                    final_word = _reduce(reduced_a + reduced_b)
+
+                words.add(final_word)
+
+    # If `words` contains `()`, filter it out before converting to list.
+    words = {w for w in words if w != ()}
+    # Convert set to list, then sort.
+    # Make sure (IDENTITY_SYMBOL,) is always at index 0.
+    list_of_words = list(words)
+    list_of_words.remove((IDENTITY_SYMBOL,))
+    # Sort remaining words: typically by length, then by content.
+    # Sorting tuples of Symbols needs a consistent key.
+    # repr(s) can give a consistent string for sorting.
+    list_of_words.sort(key=lambda w: (len(w), tuple(repr(s) for s in w)))
+    return [(IDENTITY_SYMBOL,)] + list_of_words
+
+
+def _is_zero(word: tuple[Symbol, ...]) -> bool:
+    # An empty tuple after reduction means the operator product is zero.
+    return len(word) == 0
 
-    return words
 
+def _is_identity(word: tuple[Symbol, ...]) -> bool:
+    return word == (IDENTITY_SYMBOL,)
 
-def _is_zero(word: tuple[Symbol]) -> bool:
-    return len(word) == 0
 
-
-def _is_meas(word: tuple[Symbol]) -> bool:
+def _is_meas(word: tuple[Symbol, ...]) -> bool:
+    # Expects a reduced word: (Alice_Symbol, Bob_Symbol)
     if len(word) == 2:
         s_a, s_b = word
         return s_a.player == "Alice" and s_b.player == "Bob"
-
     return False
 
 
-def _is_meas_on_one_player(word: tuple[Symbol]) -> bool:
-    return len(word) == 1 and word[0].player in {"Alice", "Bob"}
+def _is_meas_on_one_player(word: tuple[Symbol, ...]) -> bool:
+    # Expects a reduced word: (Alice_Symbol,) or (Bob_Symbol,)
+    if len(word) == 1:
+        s = word[0]
+        return s.player in ("Alice", "Bob")  # Excludes IDENTITY_SYMBOL
+    return False
 
 
+# _get_nonlocal_game_params remains the same as in npa_constraints_fix
 def _get_nonlocal_game_params(
     assemblage: dict[tuple[int, int], cvxpy.Variable], referee_dim: int = 1
 ) -> tuple[int, int, int, int]:
     a_in, b_in = max(assemblage.keys())
-    a_in = a_in + 1
-    b_in = b_in + 1
-
+    a_in += 1
+    b_in += 1
     operator = next(iter(assemblage.values()))
     a_out = int(operator.shape[0] / referee_dim)
     b_out = int(operator.shape[1] / referee_dim)
```

Comment:
[nit]: I believe you can avoid the explicit `int` case and do something like the following which uses the `//` operator:

```suggestion
    a_out = operator.shape[0] // referee_dim
    b_out = operator.shape[1] // referee_dim
```